### DISCUSSION TITLE
ProDocs: Sidebar Menu Open/Close Submenues On Click

### DISCUSSION THREAD

**Discussion by zlobica** [GetPublii/Publii#1921] [Help] [answered]
hi all, 

a few days ago I bought the ProDocs theme.

I want my left sidebar menu to work like on this page:

https://developers.google.com/search/docs/fundamentals/seo-starter-guide

please, can someone help me with this?

Thanks in advance.

**Comment by candidexmedia** [comment]
Hi! Could you provide more details about what you're hoping to achieve on your site, what's missing from the existing theme, what exactly you want from the Google example, etc?

Please provide screenshots with annotations and give a bit more explanation.

**Reply by zlobica** [reply]
hi candidexmedia

I want to have an open\close option in the sidebar navigation as it exists on the page I sent as an example.

![Screenshot_153](https://github.com/user-attachments/assets/0de0846a-88c2-43a8-90f8-0a305e07ade7)

with ProDocs theme, it is not possible to close an already open submenu.

the open submenu closes only if you click on another submenu.

**Reply by candidexmedia** [reply]
> with ProDocs theme, it is not possible to close an already open submenu.
>
> the open submenu closes only if you click on another submenu.

Gotcha... It's odd because on the mobile version of the theme, submenus can be toggled just fine. I agree that it would be a nice improvement.

I don't know how familiar you are with JavaScript, but I would inspect the current theme and see if any tweaks can be done so that you can toggle the visibility/class name of a list menu item containing sub-menu items.

Basically, the ProDocs toggleable menu items have a class called `.has-submenu`. The ones that are expanded have the class `.active`, while the ones who aren't don't have that class:

![image](https://github.com/user-attachments/assets/22ca70cd-c317-4a75-903e-cdc56d016c43)

The menu items inside of them are hidden using CSS: `display: none;` and `visibility: hidden;`

![image](https://github.com/user-attachments/assets/6b7119ce-2ca9-49fb-9e45-9ef174b15ca0)

Essentially: I would add some kind of script which adds/removes the class `.active` to toggleable menu items when they are clicked on. This should only apply to list items which already have the class name `.has-submenu`. 

If you're a fan of GPTs, start with that as a prompt, and add the JS to [Custom HTML](https://getpublii.com/docs/tools.html#customhtml) using a `<script>` tag.

If you prefer suggestions by humans, look for Codepens and the like which can provide a starting point for the toggling using vanilla JS. Here are some quick examples I found:
- https://codepen.io/tomdurkin/pen/rWZNwy
- https://codepen.io/onescu/pen/pooJBqz?anon=true&view=pen
- https://codepen.io/jacmaes/full/bGXGdX
- https://codepen.io/johndownie/pen/wvdyWXb

Maybe StackOverflow can help you out as well. Make sure to be specific about what you're trying to achieve.

**Comment by michaelfluegge** [comment]
as it is a paid theme, I would go for the Premium Support: https://marketplace.getpublii.com/my-account/

**Reply by zlobica** [reply]
They told me that I have to pay for this feature and that Premium Support does not include helping you with these kinds of things. It does not say anywhere what is included in Premium Support, and I am a little disappointed with this response and the way they do business.

**Reply by bobmitro** [reply]
> It does not say anywhere what is included in Premium Support, and I am a little disappointed with this response and the way they do business.

Hi @zlobica,

Our Terms and Conditions clearly outline the scope of Premium Support, including what is covered and what is not, specifically in point 8 regarding [Technical Support](https://getpublii.com/terms-and-conditions/#technical-support). 

The price of a theme or plugin includes three months of access to technical support, which covers general assistance, such as installation, configuration, or usage guidance. However, this support does not include custom development or modifications. Customization work is equivalent to hiring a web developer for a bespoke service, and the current cost for such services with us is €40 per hour.
Considering that the cost of our themes is roughly equal to one hour of a developer's time, it would be entirely inappropriate and unsustainable for us to offer custom development services at the price of a theme. 

If you need help with adjustments or custom solutions, we encourage you to visit our [Community Forum](https://marketplace.getpublii.com/), where you can ask for assistance from other Publii users. Many users have experience with customizations and are willing to share their solutions or offer advice.

Alternatively, you can also take advantage of our [Customization Service](https://getpublii.com/customization-service/). , where we can help tailor the theme or plugin to your specific needs based on agreed terms.

**Reply by zlobica** [reply, marked as answer]
The most common thing I heard with various WordPress themes and plugins was something like "This is a good idea, we'll try to add that feature in one of the next versions".

Never mind, I'll figure it out myself.

**Reply by michaelfluegge** [reply]
I am sorry, @bobmitro , I understood it was a feature of the theme that does not work as it should. That was why I mentioned the Premium support.

**Comment by Backspaze** [comment]
I know this discussion is closed, but I'll add a comment here and see what happens.

From my point of view this is a bug. The intended feature seems to obviously be that you can open and close the menus and sub-menus, and if you change the viewport in a desktop browser to below 900px, the menus work as on a mobile device, where you can open and close the menus without issue.

@bobmitro don't you think that this would classify as a bug? If this is not considered a bug, can you elaborate on why the menus should work differently depending on the width of the screen/viewport?
